### PR TITLE
Some cleanup of k-means internals

### PR DESCRIPTION
**Pull request by cjnolet**
This also refactors the `update centroids` logic into its own function in the detail API. Also exposing `update_centroids` function for scikit-learn pluggability.

### CODE REVIEW

**Review by divyegala — APPROVED** [review]

**Review by divyegala — APPROVED** [review]

File: cpp/include/raft/cluster/detail/kmeans_common.cuh [hunk 1]
```diff
@@ -369,6 +370,9 @@ void minClusterAndDistanceCompute(
                           true,
                           stream);
   } else {
+    // TODO: Unless pool allocator is used, passing in a workspace for this
+    // isn't really increasing performance because this needs to do a re-allocation
+    // anyways.
```
Comment (dantegd) [inline cpp/include/raft/cluster/detail/kmeans_common.cuh:375]: Not sure if there is a github issue, but the todo seems unclear, the todo is to remove the re-allocation or to remove passing workspace? Either way, a GH issue link would be useful

File: cpp/include/raft/cluster/kmeans.cuh [hunk 2]
```diff
@@ -317,14 +317,45 @@ void cluster_cost(const raft::handle_t& handle,
...
+void update_centroids(const raft::handle_t& handle,
+                      raft::device_matrix_view<const DataT, IndexT, row_major> X,
+                      raft::device_vector_view<const DataT, IndexT> sample_weights,
+                      raft::device_matrix_view<const DataT, IndexT, row_major> centroids,
+                      LabelsIterator labels,
+                      raft::device_vector_view<DataT, IndexT> weight_per_cluster,
+                      raft::device_matrix_view<DataT, IndexT, row_major> new_centroids)
+{
+  // TODO: Passing these into the algorithm doesn't really present much of a benefit
+  // because they are being resized anyways.
```
Comment (dantegd) [inline cpp/include/raft/cluster/kmeans.cuh:345]: Same comment applies

**Review by dantegd — APPROVED** [review]
Very minor comment, looks good

File: cpp/include/raft/cluster/detail/kmeans_common.cuh (hunk 1 above)
Comment (cjnolet) [reply cpp/include/raft/cluster/detail/kmeans_common.cuh:375]: The todo is to remove the workspace from the public APIs. We do have an issue for it but let me reference that in the code for completeness.

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**